flink-runtime-web/web-dashboard/src/app/pages/job/overview/backpressure/job-overview-drawer-backpressure.component.ts

```diff
@@ -41,6 +41,15 @@ export class JobOverviewDrawerBackpressureComponent implements OnInit, OnDestroy
     return node.subtask;
   }
 
+  prettyPrint(value: number): string {
+    if (isNaN(value)) {
+      return "N/A"
+    }
+    else {
+      return Math.round(value * 100) + "%";
```

Comment:
does this round up or down? (I suppose we ideally should round down so we never accidentally show 100%?

Reply:
It just simply rounds the value to the nearest integer. Why do you think we shouldn't be showing 100%?

Reply:
I was just reminded of FLINK-20424 where rounding to 100% could be misleading.
For backpressure/idling I think the distinction between 99.6% and 100% could be important (doing _something_ vs doing _nothing_). But that's just me.

Reply:
Ok, I get your point. I have two reasons why I decided to round it:

- it takes less space in the WebUI, and it's currently being displayed in one column, three ratios (back pressured / idle / busy) one after another. For example: `14% / 36% / 50%`. Adding decimal places was clogging this view a bit.
- those metrics are already approximations. For example sleeping intervals of <1ms are not measured accurately, but only asymptomatically. If you measure many intervals intervals,  this +/-1ms will average out. So even if report `100.0%` without this `Math.round()` call, it still doesn't necessarily mean usage was 100%. But 100% +/- 1ms asymptomatically. And in the worst case scenario (with every sleep interval being 0.499ms and measured as 0ms) we can have reported 0.0% usage, while in reality it should have been 49.9% (this worst case scenario is unrealistic though)

Reply:
I don't mind rounding, that is all fine. The key thing is whether to round up or down ;) If we're making a conscious decision to round to the nearest integer, then that's fine.

Reply:
👍 
I think it's fine here.